src/campaigns/util/buildMapFilters.ts

```diff
@@ -0,0 +1,138 @@
+import { Prisma } from '@prisma/client'
+import { capitalizeFirstLetter } from 'src/shared/util/strings.util'
+import { isProd } from '../mapping/campaignMap.service'
+
+interface FilterParams {
+  partyFilter?: string
+  stateFilter?: string
+  levelFilter?: string
+  resultsFilter?: boolean
+  officeFilter?: string
+  nameFilter?: string
+  forceReCalc?: boolean
+}
+
+export function buildMapFilters(
+  params: FilterParams,
+): Prisma.CampaignWhereInput[] {
+  const { partyFilter, stateFilter, levelFilter, resultsFilter, officeFilter } =
+    params
+
+  const andConditions: Prisma.CampaignWhereInput[] = []
+
+  if (partyFilter) {
+    // Prisma doesn't support case-insensitive searching inside JSON
+    const partyCondition = createJsonOrConditionString(partyFilter, ['party'])
+    if (partyCondition) {
+      andConditions.push(partyCondition)
+    }
+  }
+
+  if (stateFilter) {
+    const stateCondition = createJsonOrConditionString(stateFilter, ['state'])
+    if (stateCondition) {
+      andConditions.push(stateCondition)
+    }
+  }
+
+  if (levelFilter) {
+    const levelCondition = createJsonOrConditionString(levelFilter, [
+      'ballotLevel',
+    ])
+    if (levelCondition) {
+      andConditions.push(levelCondition)
+    }
+  }
+
+  if (resultsFilter) {
+    andConditions.push({
+      OR: [
+        { didWin: true },
+        {
+          data: {
+            path: ['hubSpotUpdates', 'election_results'],
+            equals: 'Won General',
+          },
+        },
+      ],
+    })
+  }
+
+  if (officeFilter) {
+    const officeCondition = createJsonOrConditionString(officeFilter, [
+      ['normalizedOffice'],
+      ['office'],
+      ['otherOffice'],
+    ])
+    if (officeCondition) {
+      andConditions.push(officeCondition)
+    }
+  }
+
+  if (isProd) {
+    andConditions.push({
+      data: {
+        path: ['hubSpotUpdates', 'verified_candidates'],
+        equals: 'Yes',
+      },
+    })
+  }
+
+  // Exclude campaigns without ZIP and where didWin is false
+  andConditions.push({
+    details: {
+      path: ['zip'],
+      not: { equals: null },
+    },
+    didWin: {
+      not: false,
+    },
+  })
+
+  if (isProd) {
```

Comment:
Add `TODO`s ont he `isProd`s here w/ a mention of the ticket here: https://goodparty.atlassian.net/browse/WEB-3487